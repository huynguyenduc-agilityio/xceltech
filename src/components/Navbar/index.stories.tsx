import type { Meta, StoryObj } from '@storybook/react';

// Components
import Navbar from '.';

const meta = {
  title: 'Components/Navbar',
  component: Navbar,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
} satisfies Meta<typeof Navbar>;

export default meta;

type Story = StoryObj<typeof Navbar>;

export const Default: Story = {
  args: {},
};
