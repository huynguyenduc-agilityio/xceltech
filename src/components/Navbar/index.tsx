import { useState } from 'react';

// Constants
import { USER_NAVIGATION_CONFIG } from '@/constants';

const Navbar = () => {
  const [active, setActive] = useState('Dashboard');

  return (
    <nav className="h-[122px] bg-white flex items-end justify-between">
      {/* Left side */}
      <div className=""></div>

      {/* Center - Menu */}
      <div className="flex">
        {USER_NAVIGATION_CONFIG.map((item) => (
          <button
            key={item.title}
            onClick={() => setActive(item.title)}
            className={'relative w-[160px] py-[22px] text-lg'}
          >
            {item.title}
            {active === item.title && (
              <span className="absolute left-1/2 transform -translate-x-1/2 w-[160px] h-[5px] bg-secondary bottom-0 rounded-regular"></span>
            )}
          </button>
        ))}
      </div>

      {/* TODO: Right side*/}
      <div className="flex items-center space-x-7 pb-[22px]"></div>
    </nav>
  );
};

export default Navbar;
