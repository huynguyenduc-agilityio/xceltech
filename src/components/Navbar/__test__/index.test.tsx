import { render } from '@testing-library/react';

// Components
import Navbar from '..';

const setup = () => {
  render(<Navbar />);
};

describe('Navbar component', () => {
  it('should match snapshot', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });
});
