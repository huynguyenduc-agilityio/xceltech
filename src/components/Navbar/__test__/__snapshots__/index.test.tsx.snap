// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should match snapshot 1`] = `undefined`;
